utils/zerovec/src/lib.rs

```diff
@@ -523,6 +523,53 @@ pub use zerovec_derive::make_varule;
 mod tests {
     use super::*;
     use core::mem::size_of;
+    use schemars::{gen::SchemaGenerator, JsonSchema};
+    use serde::{Deserialize, Serialize};
+    use serde_json::Value;
+
+    #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+    #[derive(JsonSchema)]
+    pub struct DataStruct<'data> {
+        #[cfg_attr(feature = "serde", serde(borrow))]
+        nums: ZeroVec<'data, u32>,
+
+        #[cfg_attr(feature = "serde", serde(borrow))]
+        chars: ZeroVec<'data, char>,
+
+        #[cfg_attr(feature = "serde", serde(borrow))]
+        strs: VarZeroVec<'data, str>,
+
+        #[cfg_attr(feature = "serde", serde(borrow))]
+        nested_numbers: VarZeroVec<'data, ZeroSlice<u32>>,
+    }
+
+    #[test]
```

Comment:
Suggestion: use [insta](https://docs.rs/insta/latest/insta/) to help generate test cases and check-in the JSON Schema output.

Reply:
Thanks, will look into it 👍